src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,536 @@
+package com.microsoft.sqlserver.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.Formatter;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.logging.StreamHandler;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+
+/**
+ * A class for testing basic NTLMv2 functionality.
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xAzureSQLDW)
+@Tag(Constants.xAzureSQLDB)
+@Tag(Constants.xAzureSQLMI)
+@Tag(Constants.NTLM)
+public class NTLMConnectionTest extends AbstractTest {
+
+    // custom formatter that contains nothing but the log message only
+    private static class CustomFormatter extends Formatter {
+        @Override
+        public String format(LogRecord record) {
+            StringBuffer sb = new StringBuffer();
+            sb.append(record.getMessage());
+            return sb.toString();
+        }
+    }
+
+    private static SQLServerDataSource dsNTLMLocal = null;
+
+    // updated connection strong with NTLM properties
+    private static String connectionStringNTLM = connectionString;
+
+    private static String serverFqdn;
+
+    private static java.util.logging.Logger ntlmLogger = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.internals.NTLMAuthentication");
+
+    private static ByteArrayOutputStream loggerContent = new ByteArrayOutputStream();
+    private static StreamHandler handler = new StreamHandler(new PrintStream(loggerContent), new CustomFormatter());
+
+    @BeforeAll
+    public static void setUp() throws Exception {
+        // remove all other logging and add our own handler
+        LogManager.getLogManager().reset();
+        ntlmLogger.setLevel(Level.FINER);
+        ntlmLogger.addHandler(handler);
+        handler.setLevel(Level.FINER);
+
+        // if these properties are defined then NTLM is desired, modify connection string accordingly
+        String domain = System.getProperty("domain");

```

Comment:
Please move the code to AbstractTest for creating connection String for NTLM.
All properties must be read at same location. They may/may not be available.